_pages/home/release-notes/checkout-libraries-release-notes.md

```diff
@@ -0,0 +1,336 @@
+- dispatch [CheckoutResetQueryEvent](#checkoutresetqueryevent) event.
+
+### CostCenterSetEvent
+
+Event is used to:
+
+- dispatch [CheckoutResetDeliveryModesEvent](#checkoutresetdeliverymodesevent) and [CheckoutResetQueryEvent](#checkoutresetqueryevent) events
+
+### PaymentTypeSetEvent
+
+Event is used to:
+
+- dispatch [CheckoutResetDeliveryModesEvent](#checkoutresetdeliverymodesevent) and [CheckoutResetQueryEvent](#checkoutresetqueryevent) events
+
+## Commands and Queries example in Checkout
+
+As mentioned, the new checkout libraries have mostly removed ngrx dependencies while adapting to our commands and queries pattern.
+
+1. Components inject facades (example: [CheckoutDeliveryAddressComponent](https://github.com/SAP/spartacus/blob/develop/feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.ts))
+
+The following describes the dependency injection to the [proxy facade](https://sap.github.io/spartacus-docs/proxy-facades/#defining-proxy-facades), where the actual implementation is provided using a dependency provider. In the case of checkout, the facade service contains the business logic executed by commands and queries.
+
+```typescript
+  constructor(
+    ...
+    protected checkoutDeliveryAddressFacade: CheckoutDeliveryAddressFacade,
+    protected checkoutDeliveryModesFacade: CheckoutDeliveryModesFacade,
+    ...
+  ) {}
+```
+
+1. Facades execute commands and queries (example: [CheckoutDeliveryAddressFacade](https://github.com/SAP/spartacus/blob/develop/feature-libs/checkout/base/core/facade/checkout-delivery-address.service.ts))
+
+Most of the time, each facade service injects both the `CommandService` and `QueryService`.
+In `CheckoutQueryFacade` we have created a checkout details query which serves as a single source of truth for the checkout state, and is updated after each change the user does during the checkout.
+
+A [command](https://sap.github.io/spartacus-docs/commands-and-queries/#commands-overview) is created in order to execute an action. In this case, it is for creating a delivery address. As seen in the snippet below, the command type is of `Command<Address, unknown>`. Address is for the input and unknown is for the output. The input for this case is required for the command to execute while the output 'may' be unknown.
+
+1. Commands can dispatch events (for other side effects)
+
+The following snippet is taken from the `createDeliveryAddressCommand`, which can be found in the source code under the [checkout-delivery-address.service.ts](https://github.com/SAP/spartacus/blob/develop/feature-libs/checkout/base/core/facade/checkout-delivery-address.service.ts#L29). After successfully creating an address, we are [dispatching events](https://sap.github.io/spartacus-docs/event-service/#dispatching-individual-events) under the tap operator in order to be caught by our event listeners to perform additional actions.
```

Comment:
```suggestion
The following snippet is taken from the `createDeliveryAddressCommand`, which can be found in the source code under the [checkout-delivery-address.service.ts](https://github.com/SAP/spartacus/blob/7bba93e0e6b75024371361b169348cfaebeb2c7b/feature-libs/checkout/base/core/facade/checkout-delivery-address.service.ts#L29). After successfully creating an address, we are [dispatching events](https://sap.github.io/spartacus-docs/event-service/#dispatching-individual-events) under the tap operator in order to be caught by our event listeners to perform additional actions.
```

Use the "copy permalink" option, as it captures the file state at the moment of writing this doc.
It will likely evolve in the future, the lines will shift, and the doc won't be correct anymore.

<img width="1252" alt="Screenshot 2022-04-25 at 12 08 13" src="https://user-images.githubusercontent.com/11561723/165068946-89ce55f0-d3e0-4eb3-8461-aefb7e134fdc.png">

Reply:
thanks for sharing!